src/data_info/services/write.clj

```diff
@@ -1,31 +1,77 @@
 (ns data-info.services.write
   (:require [clojure-commons.file-utils :as ft]
             [clojure-commons.error-codes :as ce]
+            [clojure.tools.logging :as log]
             [clojure.java.io :as io]
             [clj-jargon.item-info :as info]
             [clj-jargon.item-ops :as ops]
+            [clj-jargon.metadata :as meta]
             [clj-irods.core :as rods]
             [clj-irods.validate :refer [validate]]
+            [heuristomancer.core :as hm]
             [ring.middleware.multipart-params :as multipart]
             [otel.otel :as otel]
             [data-info.services.stat :as stat]
             [data-info.services.uuids :as uuids]
             [data-info.util.config :as cfg]
             [data-info.util.irods :as irods]
-            [data-info.util.validators :as validators]))
+            [data-info.util.validators :as validators])
+  (:import [java.io InputStream]))
+
+(defn- reset-on-close-istream
+  [^InputStream istream]
+  (proxy [InputStream] []
+    (available [] (.available istream))
+    (mark [readlimit] (.mark istream readlimit))
+    (markSupported [] (.markSupported istream))
+    (read
+      ([] (.read istream))
+      ([b] (.read istream b))
+      ([b off len] (.read istream b off len)))
+    (reset [] (.reset istream))
+    (skip [n] (.skip istream n))
+    (close []
+      (.reset istream))))
```

Comment:
This makes it so `hm/identify` doesn't actually close the `InputStream` but rather just resets it to the mark that we set a bit further down in the file (at the start of the file).